internal/product/product.service.go

```diff
@@ -22,9 +29,176 @@ func NewProductService(productUsecase ProductUsecase) ProductService {
 }
 
 func (s *productService) InsertProduct(ctx *fiber.Ctx) error {
-	_ = s.productUsecase.InsertProduct(ctx.Context())
+	var req dto.ProductRequest
+	if err := ctx.BodyParser(&req); err != nil {
+		return ctx.Status(fiber.StatusBadRequest).JSON(dto.HttpResponse{
+			Error: err.Error(),
+		})
+	}
+
+	userId, err := jwt.GetIDFromToken(ctx.Cookies("access_token"))
+
+	if err != nil {
+		return ctx.Status(fiber.StatusInternalServerError).JSON(dto.HttpResponse{
+			Error: "Cannot get UserID from access_token context: " + err.Error(),
+		})
+	}
+
+	product := entity.Product{
+		UserID:         userId,
+		Name:           req.Name,
+		IsSold:         req.IsSold,
+		Category:       req.Category,
+		Subcategory:    req.Subcategory,
+		Description:    req.Description,
+		IsVerified:     req.IsVerified,
+		Price:          req.Price,
+		ImageURL:       req.ImageURL,
+		VaccineBookURL: req.VaccineBookURL,
+	}
+
+	productRes, err := s.productUsecase.InsertProduct(ctx.Context(), &product)
```

Comment:
Should use ctx.UserContext instead not just here by should use everywhere when want to pass down the context